prometheus_flask_exporter/__init__.py

```diff
@@ -413,24 +421,42 @@ def teardown_request(exception=None):
 
             response = make_response('Exception: %s' % exception, 500)
 
+            if callable(duration_group):
+                group = duration_group(request)
+            else:
+                group = getattr(request, duration_group)
+
+            # Check if the exception was raised using a response object and use
+            # its status_code if present. This will only work for ``werkzeug.exceptions.HTTPException``
+            # and subclasses thereof. Otherwise we assume it's a 500.
+            try:
+                response = exception.get_response()
+                status_code = _to_status_code(response.status_code)
+            except AttributeError:
```

Comment:
Can we please catch all exceptions here to avoid accidentally masking the original exception with our problems?
```suggestion
            except:
```